implementation.js

```diff
@@ -73,12 +75,12 @@ if (hasSymbols) {
 	if (supportsStrIterator) {
 		iteratorSymbol = '@@iterator';
 	} else if ('Set' in global && 'Map' in global) {
-		var map = new Map();
-		map.set(1, 2);
-		var y;
-		Function('map', 'y', 'for (var x of map) {y = x}')(map, y); // eslint-disable-line no-new-func
-		if (y[0] === 1 && y[1] === 2) {
-			forOfOnly = true;
+		try {
+			var x = Function('for (var x of new Map().set(1, 2)) return x;')(); // eslint-disable-line no-new-func
```

Comment:
I'm not sure about the necessity of Map. I try to use `eval('for (var x of [0]) x;')` for now.